tests/frontend/test_engine.py

```diff
@@ -27,61 +27,54 @@
 @pytest.fixture
```

Comment:
This is more a comment on the file, ignore this particular line.

Can we rename this test to `test_local_engine.py`? The tests for the remote engines are likely to be a lot more complex with significant network imports and fixtures, so for ease of development, better to have those in separate files.

Reply:
Sure.

Reply:
Although the tests in this file mostly apply to the ABC, the other engines should pass them too.